websites/B/BeatSaver/presence.ts

```diff
@@ -0,0 +1,111 @@
+    presenceData.state = document
+      .querySelector("input.input")
+      .getAttribute("value");
+  } else if (document.location.pathname.includes("/beatmap/")) {
+    if (document.querySelector("span.tag.is-expert-plus") != null)
+      presenceData.smallImageKey = "expert_",
+        presenceData.smallImageText = "Top Diff: Expert+";
+    else if (document.querySelector("span.tag.is-expert") != null)
+      presenceData.smallImageKey = "expert",
+        presenceData.smallImageText = "Top Diff: Expert";
+    else if (document.querySelector("span.tag.is-hard") != null)
+      presenceData.smallImageKey = "hard",
+        presenceData.smallImageText = "Top Diff: Hard";
+    else if (document.querySelector("span.tag.is-normal") != null)
+      presenceData.smallImageKey = "normal",
+        presenceData.smallImageText = "Top Diff: Normal";
+    else {
+      presenceData.smallImageKey = "easy";
+      presenceData.smallImageText = "Top Diff: Easy";
+    }
+    presenceData.details = document
+      .querySelector("h1.is-size-1")
+      .textContent;
+    presenceData.state = document
+      .querySelector("h2.is-size-4")
+      .textContent;
+    presenceData.buttons = [
+      {
+        label: "View Page",
+        url: document.location.href
+      },
+      {
+        label: "View Uploader's Page",
+        url: "https://beatsaver.com" + document.querySelector("h2.is-size-4 > a").getAttribute("href")
+      }
+    ];
+  }
+  else if (document.location.pathname.includes("/uploader/"))
+    presenceData.details = "Browsing By Uploader",
+      presenceData.state = document
+        .querySelector("h1.is-size-2.has-text-weight-light.has-text-centered")
+        .textContent.split(" ")[2],
+      presenceData.buttons = [
+        {
+          label: "View Page",
+          url: document.location.href
+        }
+      ];
+  switch (document.location.pathname) {
+    case "/browse/latest":
+      presenceData.details = "Browsing By Latest";
+      break;
+    case "/browse/plays":
+      presenceData.details = "Browsing By Plays";
+      break;
+    case "/browse/downloads":
+      presenceData.details = "Browsing By Downloads";
+      break;
+    case "/browse/rating":
+      presenceData.details = "Browsing By Rating";
+      break;
+    case "/browse/hot":
+      presenceData.details = "Browsing By Hot";
+      break;
+    case "/legal/license":
+      presenceData.details = "Viewing License";
+      break;
+    case "/legal/privacy":
+      presenceData.details = "Viewing Privacy Policy";
+      break;
+    case "/legal/dmca":
+      presenceData.details = "Viewing DMCA Policy";
+      break;
+    case "/auth/login":
+      presenceData.details = "Logging In...";
+      break;
+    case "/auth/register":
+      presenceData.details = "Registering...";
+      break;
+    case "/user/upload":
+      presenceData.details = "Uploading...";
+      break;
+    case "/":
+      presenceData.details = "Viewing Home Page";
+      break;
+  }
```

Comment:
I have a feeling this part is not working correctly, please double check and upload new proof of it working.

Reply:
New proof, Every page seems to be working
![Discord_3jvtAJnHVc](https://user-images.githubusercontent.com/44418429/116813485-49e0bd80-ab54-11eb-8b51-4db58c10c471.png)
![Discord_3tlIdkOZZy](https://user-images.githubusercontent.com/44418429/116813486-4a795400-ab54-11eb-963c-49d6bce320cc.png)
![Discord_D6uuRVZsRM](https://user-images.githubusercontent.com/44418429/116813487-4a795400-ab54-11eb-8fee-3f0cf574f078.png)
![Discord_Do4BAODXYw](https://user-images.githubusercontent.com/44418429/116813488-4b11ea80-ab54-11eb-8eef-a3df218715f4.png)
![Discord_FD9U5awLN7](https://user-images.githubusercontent.com/44418429/116813489-4b11ea80-ab54-11eb-8c19-af5c955899ad.png)
![Discord_IPWlRhgqdp](https://user-images.githubusercontent.com/44418429/116813490-4b11ea80-ab54-11eb-8805-1dcc8eab931c.png)
![Discord_m9sSoBO0hX](https://user-images.githubusercontent.com/44418429/116813491-4baa8100-ab54-11eb-8743-870209caa21e.png)
![Discord_SO047YfUPg](https://user-images.githubusercontent.com/44418429/116813492-4baa8100-ab54-11eb-9ddd-2a362ddf1073.png)
![Discord_u4Kuwkdzbi](https://user-images.githubusercontent.com/44418429/116813493-4baa8100-ab54-11eb-851b-3530373b809c.png)
![Discord_UdelIDqpoo](https://user-images.githubusercontent.com/44418429/116813494-4baa8100-ab54-11eb-8aff-1d75dfb2781a.png)
![Discord_vDKJJemIGC](https://user-images.githubusercontent.com/44418429/116813495-4c431780-ab54-11eb-828d-c06d24e127e9.png)
![Discord_zPZyImOphl](https://user-images.githubusercontent.com/44418429/116813496-4c431780-ab54-11eb-9f05-760e2ab8dbf3.png)